90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -604,15 +604,33 @@ draw_snapshots() {
 # returns: nothing
 
 draw_diff() {
-  local snapshot pool diff_target mnt
+  local snapshot diff_target pool base_fs mnt
   local zfs_diff zfs_diff_PID
 
   snapshot="${1}"
   if [ -z "${snapshot}" ]; then
     zerror "snapshot is undefined"
     return 130
   fi
+
+  # if a second parameter was passed in and it's a snapshot, compare
+  # creation dates and make sure diff_target is newer than snapshot
+  if [ -n "${2}" ] && [[ "${2}" =~ @ ]]; then
```

Comment:
The second test here is potentially useless; a second parameter from the snapshot screen should always be a snapshot. There's no way to select a non-snapshot value there.